app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt

```diff
@@ -859,11 +860,11 @@ class GleanMetricsService(
 
             // setStartupMetrics is not a fast function. It does not need to be done before we can consider
             // ourselves initialized. So, let's do it, well, later.
-            setStartupMetrics()
+            setStartupMetrics(context.settings())
         }
     }
 
-    internal fun setStartupMetrics() {
+    internal fun setStartupMetrics(settings: Settings) {
```

Comment:
I think we might also want to add some docs for this function for the timing being until we consider writing telemetry documentations. In particular, I think it would be important to note why the previous approach `Addons.hasInstalledAddons.set` doesn't work and why we need to set values to a settings variable and then set telemetry like `Addons.hasInstalledAddons.set` (metric pings iirc) here.

Reply:
Sure, I'll write an explanation on why this function depends on so many shared preferences.  It won't be specific to this particular PR though.  I'll try to be general.